nagios/test/test_unit.py

```diff
@@ -0,0 +1,58 @@
+# (C) Datadog, Inc. 2018
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+
+from mock import patch
+
+from datadog_checks.nagios import Nagios
+from .common import CHECK_NAME, CUSTOM_TAGS
+
+# Random test values
+METRIC_NAME = 'nagios.metric_name'
+METRIC_VALUE = 42
+METRIC_TAGS = CUSTOM_TAGS
+METRIC_TIMESTAMP = 1337404007
+
+
+class TestGaugeWrapper:
+
+    def test_gauge_without_timestamp(self):
+        """
+        Test the 'gauge' wrapper to see if it strips the 'timestamp' arg for gauge functions that do not accept it
+        """
+
+        def timestamp_excluded_gauge(self, name, value, **kwargs):
```

Comment:
should we call these `gauge_v6` and `gauge_v5` for clarity? 
The method signature could be the same, for example
```
def gauge_v5(self, metric, value, tags=None, hostname=None, device_name=None, timestamp=None):
    assert timestamp is not None 
    ...
```
the other one would be
```
def gauge_v6(self, name, value, tags=None, hostname=None, device_name=None):
    ...
```

Reply:
Sure, that sounds like a good idea! I left it generic in case other Agents had different implementations.